cfDNA_to_mtx/mcf7_mtx_file.md

```diff
@@ -0,0 +1,239 @@
+mcf7_mtx_file
+================
+2022-06-04
+
+## Overview
+
+reference:
+<https://support.10xgenomics.com/single-cell-atac/software/pipelines/latest/output/matrices>
+
+fragment file of cfDNA is mcf7_noTx_uniq_comb_pe.sorted.bed need to
+remove last 2 columns and add a random cell barcode
+
+to replace the h5 input file, we can instead use seurat::Read10X() and
+use MEX format we need to generate the following files: ├── barcodes.tsv
+├── peaks.bed └── matrix.mtx
+
+barcodes.tsv: this is a single column file with random cell id’s in it,
+lets use 4000 repeating random strings
+
+peaks.bed: mcf7_noTx_uniq_comb_pe.sorted.bed, but use only the first 3
+columns
+
+matrix.mtx: %%MatrixMarket matrix coordinate integer general
+%metadata_json: {“format_version”: 2, “software_version”: “8000.20.5”}
+67487 4200 19929988 67437 1 2 67412 1 2 67382 1 4 67370 1 1 67366 1 2
+67361 1 4 67348 1 2
+
+row 1 meta row 2 meta row 3: 1: peaks.bed length 2: barcodes.tsv length
+3: total number of rows in mtx NOW DATA col 1: index into peaks.bed col
+2: index into barcodes.tsv col 3: counts of feature
+
+## make the fragment file
+
+``` r
+# load peak calls
+
+frag_out_file = paste0(proj_dir, "/data/data_alexis/mcf7_noTx_uniq_comb_pe_fragments.tsv")
+frag_out_gzfile = paste0(proj_dir, "/data/data_alexis/mcf7_noTx_uniq_comb_pe_fragments.tsv.gz")
+
+cfdna_peak_file = paste0(proj_dir, "/data/data_alexis/mcf7_noTx_uniq_comb_pe.sorted.bed")
+
+# subsample for speed
+cfdna_peak_df = data.frame(fread(cfdna_peak_file, header=F))
+cfdna_peak_df$diff = cfdna_peak_df$V3-cfdna_peak_df$V2
+
+# make a quick plot
+subsample_idx = sort(sample(1:nrow(cfdna_peak_df), 1000000))
+gg = ggplot(cfdna_peak_df[subsample_idx,], aes(x=diff)) + 
+        geom_histogram(binwidth=1) +
+        geom_vline(aes(xintercept=167),
+            color="blue", linetype="dashed", size=0.5) +
+        geom_vline(aes(xintercept=157),
+            color="blue", linetype="dashed", size=0.5) +
+        geom_vline(aes(xintercept=147),
+            color="blue", linetype="dashed", size=0.5) +
+        geom_vline(aes(xintercept=137),
+            color="blue", linetype="dashed", size=0.5) +
+        geom_vline(aes(xintercept=127),
+            color="blue", linetype="dashed", size=0.5) +
+        theme_bw() + labs(x ="Fragment Length", y = "Count") 
+gg
+```
+
+<img src="mcf7_mtx_file_files/figure-gfm/setup_data_files-1.png" style="display: block; margin: auto;" />
+
+``` r
+# match the correct read length
+read_len_idx = which(cfdna_peak_df$diff < 180 & cfdna_peak_df$diff > 120)
+chr_idx = which(cfdna_peak_df$V1 %in% c("chr11", "chr19"))
```

Comment:
Why are you looking at these chromosomes only?

Reply:
ah, so this is because I only care about 3 genes that are on those two chromosomes and wanted to have this run quicker.

updated the description in: https://github.com/greenelab/cfDNA_atac_demo/pull/1/commits/f36dfbde97f2bd109868748fddab2a4ce67341b6